lib/ansible/plugins/callback/default.py

```diff
@@ -328,25 +328,27 @@ def v2_playbook_on_stats(self, stats):
             t = stats.summarize(h)
 
             self._display.display(
-                u"%s : %s %s %s %s %s %s" % (
+                u"%s : %s %s %s %s %s %s %s" % (
```

Comment:
I was thinking that we should show them as subgroups, otherwise task numbers don't match up:  
``` u"%s : %s (%s) %s %s %s (%s) %s " %``
...
colorize(u'ok', t['ok'], C.COLOR_OK), 
colorize(u'rescued', t['rescued'], C.COLOR_OK),
...
 colorize(u'failed', t['failures'], C.COLOR_ERROR),
colorize(u'ignored', t['ignored'], C.COLOR_WARN),

Reply:
But `rescued` tasks are not counted as `ok` so unsure `ok=X (rescued=Y)` makes sense? Not sure `failed=X (ignored=Y)` makes sense either since those two are mutually exclusive, you can have either one or the other.

See this playbook and its recap:
```
- hosts: localhost
  gather_facts: no
  tasks:
    - name: ok1
      debug:

    - name: ok2 and changed
      debug:
      changed_when: True

    - name: skipped
      debug:
      when: False

    - block:
        - name: rescued
          fail:
      rescue:
        - name: ok3
          debug:

    - name: ok4 and ignored
      fail:
      ignore_errors: yes

    - name: failed
      fail:
```

`localhost                  : ok=4    changed=1    unreachable=0    failed=1    skipped=1    rescued=1    ignored=1`

Task numbers don't match up because `changed` and `ignored` are counted as `ok` as well.

Reply:
I seem to have confused the accounting, so `ok (ignored)` .. i assumed rescued was 'failed but rescued' but it seems to account for 'rescue tasks run'?

Reply:
If we do `ok (ignore)`, should we do also `ok (changed)`, so `ok=X (changed=Y, ignored=Z)` then? Because task number hasn't been matched up ever as `changed` tasks are counted as `ok` too.

No, `rescued` are 'failed but rescued' tasks, but they are NOT counted as `failed` so in the example above it's
```
        - name: rescued
          fail:
```
so `rescued=1` (and `failed=0` if it was just this block).

Reply:
yeah, i think that will alleviate a lot of confusion with the status of tasks